app/src/main/java/de/dreier/mytargets/base/db/migrations/Migration27.kt

```diff
@@ -0,0 +1,149 @@
+        Timber.i("Migrating DB from version 26 to 27")
+
+        database.execSQL(
+            "UPDATE RoundTemplate SET shotsPerEnd=3, endCount=10 " +
+                    "WHERE standardRoundId IN (SELECT id FROM StandardRound WHERE name ='Bray I')"
+        )
+        database.execSQL(
+            "UPDATE RoundTemplate SET shotsPerEnd=3, endCount=10 " +
+                    "WHERE standardRoundId IN (SELECT id FROM StandardRound WHERE name ='Bray II')"
+        )
+        database.execSQL(
+            "UPDATE Round SET shotsPerEnd=3, maxEndCount=10 " +
+                    "WHERE trainingId IN (SELECT t.id FROM Training as t JOIN StandardRound as s WHERE s.name ='Bray I')"
+        )
+        database.execSQL(
+            "UPDATE Round SET shotsPerEnd=3, maxEndCount=10 " +
+                    "WHERE trainingId IN (SELECT t.id FROM Training as t JOIN StandardRound as s WHERE s.name ='Bray II')"
+        )
+
+
+        database.query("SELECT `id`, `targetId`, `targetScoringStyleIndex`, `targetDiameter` FROM `Round` WHERE trainingId IN (SELECT t.id FROM Training as t JOIN StandardRound as s WHERE s.name in ('Bray I', 'Bray II'))")
+            .useEach { round ->
+                val roundId = round.getLong(0)
+                val diameter = Dimension.parse(round.getString(3))
+                val target = Target(round.getLong(1), round.getInt(2), diameter)
+
+                var index = 0
+                val ends = mutableListOf<Long>()
+                database.query("SELECT * FROM `End` WHERE `roundId` = $roundId")
+                    .useEach { end ->
+                        val end = End(
+                            id = end.getLong(0),
+                            index = end.getInt(1),
+                            roundId = end.getLong(2),
+                            exact = end.getInt(3) == 1,
+                            saveTime = LocalTime.parse(end.getString(4)),
+                            comment = end.getString(5)
+                        )
+
+                        val cursor = database.query("SELECT `fileName` FROM `EndImage` WHERE `endId` = ${end.id}")
+                        var fileName: String? = null
+                        if (cursor.moveToFirst()) {
+                            fileName = cursor.getString(0)
+                        }
+
+                        ends.add(end.id)
+
+                        addEnd(database, target, end, fileName, index++)
+                        addEnd(database, target, end, fileName, index++)
+
+                        deleteEndImage(database, end.id) // trigger?
+                        deleteShots(database, end.id)
+                    }
+
+                deleteEnds(database, ends.joinToString())
+            }
+    }
+
+    fun addEnd(database: SupportSQLiteDatabase, target: Target, end: End, fileName: String?, index: Int) {
+        val shots = mutableListOf<Shot>()
+        database.query("SELECT `id`, `index`, `scoringRing` FROM Shot WHERE endId = ${end.id} LIMIT 3")
```

Comment:
This is a bit dangerous as the result of LIMIT is not deterministic if no `SORT BY` clause is given. (Should be sorted by id)